### ISSUE TITLE
MINSIGSTKSZ may not be constant

### ISSUE THREAD

**Issue by jamesjer** [libsemigroups/libsemigroups#271]
In the not-yet-released glibc 2.34, `MINSIGSTKSZ` is no longer a constant, but instead expands to `sysconf(_SC_SIGSTKSZ)`.  This leads to a test failure:
```
In file included from /usr/include/signal.h:315,
                 from tests/catch.hpp:7712,
                 from tests/test-main.cpp:26:
tests/catch.hpp:10453:58: error: call to non-'constexpr' function 'long int sysconf(int)'
10453 |     static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
      |                                                          ^~~~~~~~~~~
In file included from /usr/include/bits/sigstksz.h:24,
                 from /usr/include/signal.h:315,
                 from tests/catch.hpp:7712,
                 from tests/test-main.cpp:26:
/usr/include/unistd.h:630:17: note: 'long int sysconf(int)' declared here
  630 | extern long int sysconf (int __name) __THROW;
      |                 ^~~~~~~
In file included from tests/test-main.cpp:26:
tests/catch.hpp:10512:45: error: size of array 'altStackMem' is not an integral constant-expression
10512 |     char FatalConditionHandler::altStackMem[sigStackSize] = {};
      |                                             ^~~~~~~~~~~~
make[1]: *** [Makefile:2745: tests/test-main.o] Error 1
```

This is causing libsemigroups builds to fail in Fedora Rawhide.

**Comment by james-d-mitchell** [comment]
Thanks @jamesjer! Looks like an issue with catch, I wonder is this resolved by updating to the latest version of Catch (v2.13.2) from the version currently used in `libsemigroups` (v2.9.2)?

**Comment by james-d-mitchell** [comment]
Looks like updating to Catch v2.13.4 didn't change this at all.

**Comment by james-d-mitchell** [comment]
See for more details:

https://github.com/catchorg/Catch2/issues/2178

**Comment by james-d-mitchell** [comment]
Resolved in v2.1.0 or possibly earlier
